metals/src/main/scala/scala/meta/internal/metals/BuildTargetInfo.scala

```diff
@@ -0,0 +1,306 @@
+          info += "  [BLANK]"
+      })
+      detail.scala.foreach(scala => {
+        info += ""
+        info += "Scalac Options"
+        if (scala.scalaBinaryVersion.startsWith("3"))
+          info += "  compile - https://docs.scala-lang.org/scala3/guides/migration/options-new.html"
+        else
+          info += "  compile - https://docs.scala-lang.org/overviews/compiler-options/index.html#Standard_Settings"
+        info += "  "
+        if (scala.options.nonEmpty)
+          scala.options.foreach(f =>
+            info += s"  ${if (f.isEmpty) "[BLANK]" else f}"
+          )
+        else
+          info += "  [BLANK]"
+
+        info += ""
+        info += "Scala Version"
+        info += s"  ${scala.scalaVersion}"
+        info += ""
+        info += "Scala Binary Version"
+        info += s"  ${scala.scalaBinaryVersion}"
+        info += ""
+        info += "Scala Platform"
+        info += s"  ${scala.scalaPlatform}"
+        scala.jvmVersion.foreach(jvmVersion => {
+          info += ""
+          info += "JVM Version"
+          info += s"  $jvmVersion"
+        })
+        scala.jvmHome.foreach(jvmHome => {
+          info += ""
+          info += "JVM Home"
+          info += s"  $jvmHome"
+        })
+      })
+      info += ""
+      info += "Base Directory"
+      info += s"  ${detail.common.baseDirectory}"
+      info += ""
+      info += "Source Directories"
+      if (detail.common.sources.nonEmpty)
+        detail.common.sources.foreach(f => info += s"  $f")
+      else
+        info += "  NONE"
+      if (detail.sameJavaScalaClassesDir)
+        detail.java.foreach(java => {
+          info += ""
+          info += "Classes Directory"
+          info += s"  ${java.classesDirectory}"
+        })
+      else {
+        detail.java.foreach(java => {
+          info += ""
+          info += "Java Classes Directory"
+          info += s"  ${java.classesDirectory}"
+        })
+        detail.scala.foreach(scala => {
+          info += ""
+          info += "Scala Classes Directory"
+          info += s"  ${scala.classesDirectory}"
+        })
+      }
+      if (detail.sameJavaScalaClasspaths)
+        detail.java.foreach(java => {
+          info += ""
+          info += "Classpath"
+          addClassPath(info, java.classPath)
+        })
+      else {
+        detail.java.foreach(java => {
+          info += ""
+          info += "Java Classpath"
+          addClassPath(info, java.classPath)
+        })
+        detail.scala.foreach(scala => {
+          info += ""
+          info += "Scala Classpath"
+          addClassPath(info, scala.classPath)
+        })
+      }
+      info += ""
+      info.mkString(System.lineSeparator())
+    })
+  }
+
+  private def addClassPath(
+      info: ListBuffer[String],
+      classPath: List[Path]
+  ): Unit = {
+    def convertToName(path: Path): Path = {
+      if (path.toFile.isFile)
+        path.getFileName()
+      else
+        path
+    }
+    if (classPath.nonEmpty) {
```

Comment:
I'm recommend breaking this up a bit. We have 3 nested ifs in here that can be tricky to follow. Breaking those out to functions might make this easier to read.